packages/wrangler/src/__tests__/pipelines.test.ts

```diff
@@ -0,0 +1,571 @@
+							messages: [],
+							result: pipeline,
+						},
+						{ status }
+					);
+				},
+				{ once: true }
+			)
+		);
+		return requests;
+	}
+
+	function mockDeleteRequest(
+		name: string,
+		status: number = 200,
+		error?: object
+	) {
+		const requests = { count: 0 };
+		msw.use(
+			http.delete(
+				"*/accounts/:accountId/pipelines/:name",
+				async ({ params }) => {
+					requests.count++;
+					expect(params.accountId).toEqual("some-account-id");
+					expect(params.name).toEqual(name);
+
+					return HttpResponse.json(
+						{
+							success: !error,
+							errors: error ? [error] : [],
+							messages: [],
+							result: null,
+						},
+						{ status }
+					);
+				},
+				{ once: true }
+			)
+		);
+		return requests;
+	}
+	beforeAll(() => {
+		__testSkipDelays();
+	});
+
+	it("shows usage details", async () => {
+		await runWrangler("pipelines");
+		await endEventLoop();
+
+		expect(std.err).toMatchInlineSnapshot(`""`);
+		expect(std.out).toMatchInlineSnapshot(`
+          "wrangler pipelines
+
+          🚰 Manage Worker Pipelines
+
+
+          COMMANDS
+            wrangler pipelines create <pipeline>  Create a new pipeline
+            wrangler pipelines list               List current pipelines
+            wrangler pipelines show <pipeline>    Show a pipeline configuration
+            wrangler pipelines update <pipeline>  Update a pipeline
+            wrangler pipelines delete <pipeline>  Delete a pipeline
+
+          GLOBAL FLAGS
+            -j, --experimental-json-config  Experimental: support wrangler.json  [boolean]
+            -c, --config                    Path to .toml configuration file  [string]
+            -e, --env                       Environment to use for operations and .env files  [string]
+            -h, --help                      Show help  [boolean]
+            -v, --version                   Show version number  [boolean]"
+        `);
+	});
+
+	it("create - should create a pipeline", async () => {
+		const tokenReq = mockCreateR2Token("test-bucket");
+		const requests = mockCreateRequest("my-pipeline");
+		await runWrangler("pipelines create my-pipeline --r2 test-bucket");
+
+		expect(tokenReq.count).toEqual(3);
+		expect(requests.count).toEqual(1);
+	});
+
+	it("create - should create a pipeline with explicit credentials", async () => {
+		const requests = mockCreateRequest("my-pipeline");
+		await runWrangler(
+			"pipelines create my-pipeline --r2 test-bucket --access-key-id my-key --secret-access-key my-secret"
+		);
+		expect(requests.count).toEqual(1);
+	});
+
+	it("create - should fail a missing bucket", async () => {
+		const requests = mockCreeatR2TokenFailure("bad-bucket");
+		await expect(
+			runWrangler("pipelines create bad-pipeline --r2 bad-bucket")
+		).rejects.toThrowError();
+
+		await endEventLoop();
+
+		expect(std.err).toMatchInlineSnapshot(`
+			"[31mX [41;31m[[41;97mERROR[41;31m][0m [1mThe R2 bucket [bad-bucket] doesn't exist[0m
```

Comment:
nit: consider using normalizingOutput to remove the ANSI codes before snapshotting